packages/frontend/components/scaffold-eth/Contract/InputUI.tsx

```diff
@@ -0,0 +1,102 @@
+import { ethers } from "ethers";
+import { FunctionFragment } from "ethers/lib/utils";
+import React, { Dispatch, FC, ReactElement, SetStateAction } from "react";
+import Address from "../Address";
+
+type ParamType = {
+  name: string | null;
+  type: string;
+};
+
+interface IInputUI {
+  setForm: Dispatch<SetStateAction<Record<string, any>>>;
+  form: Record<string, any>;
+  stateObjectKey: string;
+  paramType: ParamType;
+  functionFragment: FunctionFragment;
+}
+
+// TODO use AddressInput component instead
```

Comment:
Currently this component uses `Address` component to show input address, but we should use the `AddressInput` component instead, we need to update `AddressInput` to take in `onChange` and `value` as prop similar to abi.ninja's [AddressInput.jsx](https://github.com/carletex/abi.ninja/blob/master/packages/react-app/src/components/AddressInput.jsx) component

Reply:
Yeah, cool! I talked about this with Naim while working on the Address component: https://github.com/scaffold-eth/se-2/pull/11#discussion_r999625592

> Also, let's not worry too much, we can adapt in the future depending on the needs

I think it's time to adapt :) I'm working on this. Will update.

Reply:
@technophile-04 Just updated the `<AddressInput>` component on 19c5e6bad0fb353aed4e79478fca75442e3fa728.

You can use it with something like:

```tsx
      <AddressInput
        name={stateObjectKey}
        value={form[stateObjectKey]}
        onChange={(value): void => {
          const formUpdate = { ...form };
          formUpdate[stateObjectKey] = value;
          setForm(formUpdate);
        }}
      />
```

Let me know if it works for you.